PR: https://github.com/odoo/odoo/pull/104223

From: 77f9ff50db3cdb88397d0b1cc7042c772d0d417b
From: Florian Charlier
Date: 2023-06-30 21:37:50

Breaking data model changes scores: del:1 + add:0, change matches:
-    _inherit = 'res.company'

Total Changes: 41

[REF] *: use onboarding module

* = account{_payment}, base, onboarding, payment{_stripe},
sale{_management}, web, website_sale

Use the dedicated onboarding module introduced in 16.0 instead of
the res.company model to store onboarding progress.

It allows
 * onboarding steps to be reused across panels
 * to support steps that should be completed per-database or per-company
 * to clean the res.company model from many fields and methods,
 * to remove many views, controllers, actions

Module-specific notes:
* account: We also clean the remaining two steps that are not
part of an accounting panel but make the most sense to be kept here.
* account_payment: Following 8e4e8eb8, the payment provider step is
added to the invoicing onboarding panel. We apply this change here too.
Also impacts the website_sale_dashboard panel (see related ENT PR).
(The "sale tax" one is currently used for to the website sale dashboard).
* payment: Note that the step was already not part of an onboarding
panel within this module.
* website_sale: We clean
  * a field not used (The website_sale dashboard onboarding panel used
  the payment_provider_onboarding_state field).
  * a method that was only called from website_sale_dashboard, so it is
  moved there. See related ENT PR.

Includes a few tests.

Moving views/templates/styling, as well as cleaning residual onboarding-related fields and methods in base, including populate.

This also includes restoring the "onboarding_complete" overlay panel
animating it to disappear after a few seconds so that it doesn't hide
text and block buttons to re-open steps.

Task-3025136

Part-of: odoo/odoo#104223

================================= pseudo patch: =================================

--- a/addons/account_payment/models/__init__.py
+++ b/addons/account_payment/models/__init__.py
@@ -5,6 +5,6 @@ from . import account_move
 from . import account_payment
 from . import account_payment_method
 from . import account_payment_method_line
+from . import onboarding_onboarding_step
 from . import payment_provider
 from . import payment_transaction
-from . import res_company

--- a/None
+++ b/addons/account_payment/models/onboarding_onboarding_step.py
@@ -0,0 +1,20 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+
+
+class OnboardingStep(models.Model):
+    _inherit = 'onboarding.onboarding.step'
+
+    @api.model
+    def action_open_step_payment_provider(self):
+        self.env.company.payment_onboarding_payment_method = 'stripe'
+        menu = self.env.ref('account_payment.payment_provider_menu', raise_if_not_found=False)
+        menu_id = menu.id if menu else None
+        return self.env.company._run_payment_onboarding_step(menu_id)
+
+    @api.model
+    def action_validate_step_payment_provider(self):
+        validation_response = super().action_validate_step_payment_provider()
+        self.action_validate_step("account_payment.onboarding_onboarding_step_payment_provider")
+        return validation_response

--- a/addons/account_payment/models/res_company.py
+++ b/None
@@ -1,19 +0,0 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import api, models
-
-
-class ResCompany(models.Model):
-    _inherit = 'res.company'
-
-    @api.model
-    def action_open_payment_onboarding(self):
-        self.env.company.payment_onboarding_payment_method = 'stripe'
-        menu = self.env.ref('account_payment.payment_provider_menu', raise_if_not_found=False)
-        menu_id = menu and menu.id
-        return self._run_payment_onboarding_step(menu_id)
-
-    def get_account_invoice_onboarding_steps_states_names(self):
-        """ Override of `account` to add the state of the payment onboarding step. """
-        steps = super().get_account_invoice_onboarding_steps_states_names()
-        return steps + ['payment_provider_onboarding_state']
